src/firefly/java/edu/caltech/ipac/visualize/plot/PixelValue.java

```diff
@@ -152,7 +153,14 @@ private static int getBytePerPixel(int bitpix) {
 		};
 	}
 
-	private static boolean isInt(DirectFitsAccessData h) { return h.bitpix()>0;}
+	private static boolean isInt(DirectFitsAccessData h) {
+		// From FITS 4.00 (https://fits.gsfc.nasa.gov/standard40/fits_standard40aa-le.pdf), p. 14-15, Table 11 and section 4.4.2.5
+		return (h.bitpix()>0 && h.bScale()==1.0 && (h.bZero()==0
+				|| (h.bitpix()==8 && h.bZero()==-128.0)
+				|| (h.bitpix()==16 && h.bZero()==32768.0)
+				|| (h.bitpix()==32 && h.bZero()==2147483648.0)
```

Comment:
This is a little hard to read. You should put a comment describing what you are checking for.